src/apps/js_v8/tmpl/template.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "v8.h"
+#include "v8_runner.h"
+#include "v8_util.h"
+
+namespace
+{
+  template <class T>
+  static v8::Local<v8::Template> get_cached_template(v8::Isolate* isolate)
+  {
+    ccf::V8Isolate::TemplateCache& cache =
+      ccf::V8Isolate::Data::Get(isolate)->get_template_cache();
+    if (!cache.has(T::NAME))
+    {
+      v8::Local<v8::Template> raw_template = T::create_template(isolate);
+      cache.set(T::NAME, raw_template);
+    }
+    v8::Local<v8::Template> tmpl = cache.get(T::NAME);
+    return tmpl;
+  }
+}
+
+namespace ccf::v8_tmpl
+{
+  template <class T>
+  static v8::Local<v8::ObjectTemplate> get_cached_object_template(
+    v8::Isolate* isolate)
+  {
+    v8::Local<v8::Template> tmpl = get_cached_template<T>(isolate);
+    return tmpl.As<v8::ObjectTemplate>();
+  }
+
+  template <class T>
+  void set_internal_field_count(v8::Local<v8::ObjectTemplate> tmpl)
+  {
+    static_assert(std::is_enum_v<T>, "T must be an enum class");
+    tmpl->SetInternalFieldCount(sizeof(T));
```

Comment:
That's not right, or at least it's very different from what you were doing before. Mostly this will set the InternalFieldCount to 4, unless you have some quite large enums, since this is the number of bytes clang 10 uses by default for an enum class

Try:

```
#include <iostream>

enum class Foo
{
    Bar
};

int main()
{
    std::cout << sizeof(Foo) << std::endl;
}
```

Reply:
I went for an `END` marker member and replaced `set_internal_field` with a new `set_internal_fields` which uses a `std::array` with the static size templated based on `T::END`.